src/plugins/initiatives/discourse.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+import {DomHandler, DomUtils, Parser} from "htmlparser2";
+
+import {type Topic, type Post} from "../discourse/fetch";
+import {type Initiative, type URL} from "./initiative";
+import {topicAddress} from "../discourse/address";
+
+type HeaderToURLsMap = Map<string, $ReadOnlyArray<URL>>;
+
+export async function initiativeFromDiscourseTracker(
+  serverUrl: string,
+  topic: Topic,
+  firstPost: Post
+): Promise<Initiative> {
+  if (firstPost.topicId !== topic.id) {
+    throw new Error("Post is from a different topic");
+  }
+
+  if (firstPost.indexWithinTopic !== 1) {
+    throw new Error("Post is not the first post in the topic");
+  }
+
+  const {title} = topic;
+  const {timestampMs} = firstPost;
+  const tracker = topicAddress(serverUrl, topic.id);
+
+  const map = await groupURLsByHeader(firstPost.cooked);
```

Comment:
I've included the type, though still use a shorthand name, otherwise prettier will indent some calls but not others. Making it much less readable imo.